PWGHF/D2H/TableProducer/candidateCreatorBplusReduced.cxx

```diff
@@ -45,19 +47,23 @@ struct HfCandidateCreatorBplusReduced {
   Configurable<double> minParamChange{"minParamChange", 1.e-3, "stop iterations if largest change of any B+ is smaller than this"};
   Configurable<double> minRelChi2Change{"minRelChi2Change", 0.9, "stop iterations is chi2/chi2old > this"};
   // selection
-  Configurable<double> invMassWindowBplus{"invMassWindowBplus", 0.3, "invariant-mass window for B+ candidates"};
+  Configurable<double> invMassWindowD0PiTolerance{"invMassWindowD0PiTolerance", 0.01, "invariant-mass window tolerance for D0Pi pair preselections (GeV/c2)"};
+  // variable that will store the value of invMassWindowD0Pi (defined in dataCreatorD0PiReduced.cxx)
+  float myInvMassWindowD0Pi{1.};
 
-  double massPi = RecoDecay::getMassPDG(kPiPlus);
-  double massD0 = RecoDecay::getMassPDG(pdg::Code::kD0);
-  double massBplus = RecoDecay::getMassPDG(pdg::Code::kBPlus);
-  double massD0Pi{0.};
+  // O2DatabasePDG service
+  Service<o2::framework::O2DatabasePDG> pdg;
+
+  double massPi{0.};
+  double massD0{0.};
+  double massBplus{0.};
   double bz{0.};
 
   // Fitter for B vertex (2-prong vertex filter)
   o2::vertexing::DCAFitterN<2> df2;
 
-  Preslice<aod::HfCand2ProngReduced> candsDPerCollision = hf_track_index_reduced::hfReducedCollisionId;
-  Preslice<aod::HfTracksReduced> tracksPionPerCollision = hf_track_index_reduced::hfReducedCollisionId;
+  Preslice<soa::Join<aod::HfRed2Prongs, aod::HfRed2ProngsCov>> candsDPerCollision = hf_track_index_reduced::hfRedCollisionId;
```

Comment:
In the `candidateCreatorBplus.cxx` there D-pi candidates are filtered accounting for the response of the `candidateSelectorD0` (see here: https://github.com/AliceO2Group/O2Physics/blob/c8f778a304ee9723d1d05a95dd47a240fded21e6/PWGHF/TableProducer/candidateCreatorBplus.cxx#L96) while in your case you are not doing it. This is because you already do the D0 selections in `dataCreatorD0PiReduced`, right?

Reply:
Exactly, this part of the Standard creator is already implemented at the DPi pair creation level of the Split workflow.